test/e2e/lib/broker.go

```diff
@@ -70,6 +70,14 @@ func (a BrokerMetricAssertion) Assert(client *monitoring.MetricClient) error {
 		if err != nil {
 			return fmt.Errorf("metric has invalid response code label: %v", ts.GetMetric())
 		}
+
+		// Workarounds to reduce test flakiness caused by sender pod retry sending events (which will cause unexpected response code 404, 503 and 500).
```

Comment:
I think it's the unexpected response code that triggers the sender to retry, not the other way around. We can say something like "ignore undesired response code in metric count comparison due to flakiness in broker, the sender pod will retry until event is successfully received".